### PR TITLE
add function resource profile to conversion from function v1alpha2 to…

### PR DESCRIPTION
**Pull request by anoipm**
**Description**

Changes proposed in this pull request:

- add function resource profile to conversion from function v1alpha2 to v1alpha1

**Related issue(s)**
See also: #15288 (PR does only a small part of this task)

### CODE REVIEW

File: components/function-controller/internal/webhook/conversion_webhook.go [hunk 1]
```diff
@@ -327,8 +339,16 @@ func convertResourcesV1Alpha2ToV1Alpha1(in *serverlessv1alpha2.Function, out *se
 	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Build != nil && in.Spec.ResourceConfiguration.Build.Resources != nil {
 		out.Spec.BuildResources = *in.Spec.ResourceConfiguration.Build.Resources
 	}
-	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Function != nil && in.Spec.ResourceConfiguration.Function.Resources != nil {
-		out.Spec.Resources = *in.Spec.ResourceConfiguration.Function.Resources
+	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Function != nil {
+		if in.Spec.ResourceConfiguration.Function.Resources != nil {
+			out.Spec.Resources = *in.Spec.ResourceConfiguration.Function.Resources
+		}
+		if in.Spec.ResourceConfiguration.Function.Profile != "" {
```
Comment (dbadura) [inline components/function-controller/internal/webhook/conversion_webhook.go:346]: We can wrap it in function `convertResourceProfileV1alpha2ToV1Alpha1`

Comment (anoipm) [reply components/function-controller/internal/webhook/conversion_webhook.go]: changed

**Review by dbadura — DISMISSED** [review]

**Review by dbadura — APPROVED** [review]

**Pull request merged**